### ISSUE TITLE
Product-Area-Here: short description of bug here

### ISSUE THREAD

**Issue by djukic-marko** [grafana/grafana#68781]
### 
# What went wrong?

Hi, 

We are using grafana hosted in a container in a host network configuration.
That means grafana is running on HTTP port 3000.
However we have AWS LB in front of it with SSL certificate, that forwards traffic to grafana instance.

We have set up root_url in grafana with full FQDN including https.

When using the share button from dashboards, links for sharing are formed correctly, however when we switch the toggle to "short url", link no longer respects root_url parameter and includes HTTP and port 3000. As if root_url configuration is ignored in that case.
Also, when sharing something from explore tab, short links are always used without an option for a full URL. That means that the wrong link is formed every time.

Im not sure if this is a bug or a configuration error.

### 
# How do we reproduce it?

Configure root_url parameter and use the short link. The short link will not respect root_url parameter

### 
# What Grafana version are you using?

9.4.7-ubuntu

### 
### Grafana Platform?

Docker

### User's OS?

Ubuntu

### User's Browser?

Chrome latest

### Is this a Regression?

None

**Comment by tonypowa** [comment]
hi @djukic-marko 

thank you for creating this issue

could you safely share the [[server]](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#server) part of your configuration ?

tagging the explore and dashboard squads for review  (please reroute if needed)

**Comment by djukic-marko** [comment]
Sure thing, here it is:

`
#################################### Server ####################################
[server]
# Protocol (http, https, h2, socket)
;protocol = http

# The ip address to bind to, empty will bind to all interfaces
;http_addr =

# The http port  to use
http_port = 3000

# The public facing domain name used to access grafana from a browser
domain = grafana.ourdomain.com

# Redirect to correct domain if host header does not match domain
# Prevents DNS rebinding attacks
;enforce_domain = false

# The full public facing url you use in browser, used for redirects and emails
# If you use reverse proxy and sub path specify full url (with sub path)
;root_url = %(protocol)s://%(domain)s:%(http_port)s/
root_url = https://grafana.ourdomain.com

# Serve Grafana from subpath specified in `root_url` setting. By default it is set to `false` for compatibility reasons.
;serve_from_sub_path = false

# Log web requests
;router_logging = false

# the path relative working path
;static_root_path = public

# enable gzip
;enable_gzip = false

# https certs & key file
;cert_file =
;cert_key =

# Unix socket gid
# Changing the gid of a file without privileges requires that the target group is in the group of the process and that the process is the file owner
# It is recommended to set the gid as http server user gid
# Not set when the value is -1
;socket_gid =

# Unix socket mode
;socket_mode =

# Unix socket path
;socket =

# CDN Url
;cdn_url =

# Sets the maximum time using a duration format (5s/5m/5ms) before timing out read of an incoming request and closing idle connections.
# `0` means there is no timeout for reading the request.
;read_timeout = 0

# This setting enables you to specify additional headers that the server adds to HTTP(S) responses.
[server.custom_response_headers]
#exampleHeader1 = exampleValue1
#exampleHeader2 = exampleValue2
`

**Comment by natellium** [comment]
Interestingly enough, while trying to reproduce the issue I had the opposite result. The share link only shows the root_url config when selecting the short URL. @evictorero as you're in triage next week, could you check what behaviour you have?

https://github.com/grafana/grafana/assets/74534993/49b427d0-5f7e-456b-8e16-8a948786f15c

**Comment by djukic-marko** [comment]
Yeah, I tested again and I got the opposite result from you. Just to be sure, you are running in docker HTTP and SSL termination is in front of the container?
Anyways, even if it works opposite for you it still looks like a bug, or I forgot to configure something?

**Comment by evictorero** [comment]
I tested it and work as expected in `9.4.7`, when I switch the toggle I receive a shorten URL with the `root_url` configured in my `config.ini`.
The normal URL is created by the frontend and the short URL is created by the backend. So every time you create a shortened URL a backend request is made. Can you check the result of this request using the `dev tools->network tab` in your browser?

**Comment by djukic-marko** [comment]
I tested again and got the same (wrong) result. 
Checked my docker image version, as well as the config files. Nothing has changed.
Decided to restart the container one more time, and after the container restarts, It seems to be working right.
I'm pretty sure I restarted the container after applying the configuration the first time, but I guess I did not :(.
I also checked the network response from API, and it's giving the right response with the correct TLD and port.

This is not an issue anymore, but it's interesting how Natellium got similar behavior to occur.

**Comment by evictorero** [comment]
Nice! I'm glad it's not an issue. @natellium case it's not actually the opposite, she got it working as expected but because she didn't configure a different URL in the local instance the case appeared to be the opposite.
